delfin/drivers/hpe/hpe_3par/ssh_handler.py

```diff
@@ -421,17 +422,22 @@ def parse_node_table(self, cols_size, titles_size, str_info, obj_list,
                 obj_list.append(obj_model)
         return obj_list
 
-    def parse_node_cpu(self, cols_size, titles_size, str_info, obj_map):
+    def parse_node_cpu(self, cols_size, titles_size, str_info, obj_map,
+                       titles):
         if cols_size >= titles_size:
-            node_id = str_info[0]
-            cpu_info = str_info[4]
-            if obj_map.get(node_id):
-                obj_map[node_id][cpu_info] = obj_map.get(node_id).get(
-                    cpu_info, 0) + 1
+            if 'Cores' in titles:
+                node_id = str_info[0]
+                cpu_info = ' '.join(str_info[5:])
+                cpu_map = obj_map.setdefault(node_id, {})
+                cpu_map[cpu_info] = int(str_info[2])
             else:
-                cpu_info_map = {}
-                cpu_info_map[cpu_info] = 1
-                obj_map[node_id] = cpu_info_map
+                node_id = str_info[0]
+                cpu_info = str_info[4]
+                if obj_map.get(node_id):
```

Comment:
We should always try to avoid using `if ... esle ...` , it's recommended to develop like:
```python
cpu_map = obj_map.get(node_id, {})
cpu_map[cpu_info] = cpu_map.get(cpu_info, 0) + 1
```

Reply:
modified